src/exporter.js

```diff
@@ -0,0 +1,104 @@
+// Copyright 2020 Stanford University see LICENSE for license
+
+import honeybadger from 'honeybadger'
+import config from 'config'
+import fs from 'fs'
+import { query } from './sinopia_client'
+
+const Honeybadger = honeybadger.configure({
+  apiKey: process.env.HONEYBADGER_API_KEY
+})
+
+const getDateString = () => {
+  return (new Date()).toISOString()
+}
+
+const initAndGetSavePath = (groupName, exportBasePath) => {
+  let dateTag
+  if (!exportBasePath) {
+    exportBasePath = config.get('exportBasePath')
+    dateTag = getDateString()
+  }
+
+  let savePathString = [exportBasePath, groupName].join('/')
+  if (dateTag)
+    savePathString = [savePathString, dateTag].join('_');
+
+  fs.mkdirSync(savePathString, { recursive: true })
+  return savePathString
+}
+
+const reportError = (errorObject, consoleErrMessage = null) => {
+  Honeybadger.notify(errorObject, {
+    context: {
+      argv: process.argv,
+      sinopiaApiBasePath: config.get('sinopia_api.basePath')
+    }
+  })
+
+  if (consoleErrMessage)
+    console.error(consoleErrMessage)
+}
+
+export const exportGroup = async (groupName, containingDir = '') => {
+  console.info(`beginning export of RDF from group: ${groupName}`)
+
+  const savePathString = initAndGetSavePath(groupName, containingDir)
+  let done = false
+  let uri = [config.get('sinopia_api.basePath'), `/resource/?limit=${config.get('sinopia_api.query_limit')}&start=0&group=`, groupName].join('')
+
+  while (!done) {
```

Comment:
How do you exit the loop (or than L71)?

Reply:
Yes, the line on 71. I could set done in the catch statements if that is a concern, but we can't break inside of an arrow function (results in `unsyntactic break` error.

Reply:
Shouldn't it stop when it runs out of results?

Reply:
It did, if there is no next link or the results are less then expected. I updated this with a do/while for some clarity, hope that helps @justinlittman